egg-herbie/src/lib.rs

```diff
@@ -254,6 +275,136 @@ pub unsafe extern "C" fn egraph_get_simplest(
     })
 }
 
+fn get_op_str(n: &Math) -> Option<String> {
+    match n {
+        Math::Add(_) => Some(String::from("+")),
+        Math::Sub(_) => Some(String::from("-")),
+        Math::Mul(_) => Some(String::from("*")),
+        Math::Div(_) => Some(String::from("/")),
+        Math::Pow(_) => Some(String::from("pow")),
+        Math::Neg(_) => Some(String::from("neg")),
+        Math::Sqrt(_) => Some(String::from("sqrt")),
+        Math::Fabs(_) => Some(String::from("fabs")),
+        Math::Ceil(_) => Some(String::from("ceil")),
+        Math::Floor(_) => Some(String::from("floor")),
+        Math::Round(_) => Some(String::from("round")),
+        Math::Log(_) => Some(String::from("log")),
+        Math::Cbrt(_) => Some(String::from("cbrt")),
+        Math::Other(s, _) => Some(s.to_string()),
+        _ => None,
+    }
+}
+
+fn variant_get_simplest(
```

Comment:
why do we cache? The extractor should already do that right?

Reply:
Does it cache entire recexprs?

Reply:
Ok regardless, I'm using the an Id -> RecExpr cache to get around the reference requirement for `to_recexpr`

Reply:
Ah okay
It doesn't cache full recexprs, I think it just stores the best node per eclass